From 1bbb1889ab7b4ec74689189d324a4fc4112959f5 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
Date: Tue, 23 Feb 2010 13:00:01 -0600
Subject: [PATCH] ENGR00121069: MX51: Complete the constraints for the various regulators.

Completed intialization of mc13892 regulators. This was causing the
regulator API to leave many regulators enabled at boot and also print
error messages at boot.

Signed-off-by: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
---
 arch/arm/mach-mx51/mx51_babbage.c              |    1 -
 arch/arm/mach-mx51/mx51_babbage_pmic_mc13892.c |   12 ++++++------
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx51/mx51_babbage.c b/arch/arm/mach-mx51/mx51_babbage.c
index 715fecf..43ab956 100644
--- a/arch/arm/mach-mx51/mx51_babbage.c
+++ b/arch/arm/mach-mx51/mx51_babbage.c
@@ -183,7 +183,6 @@ static struct mxc_srtc_platform_data srtc_data = {
 
 static struct tve_platform_data tve_data = {
 	.dac_reg = "VVIDEO",
-	.dig_reg = "VDIG",
 };
 
 static struct mxc_dvfs_platform_data dvfs_core_data = {
diff --git a/arch/arm/mach-mx51/mx51_babbage_pmic_mc13892.c b/arch/arm/mach-mx51/mx51_babbage_pmic_mc13892.c
index 70be52f..df57517 100644
--- a/arch/arm/mach-mx51/mx51_babbage_pmic_mc13892.c
+++ b/arch/arm/mach-mx51/mx51_babbage_pmic_mc13892.c
@@ -99,11 +99,6 @@ static struct regulator_consumer_supply vdig_consumers[] = {
 		.supply = "VDDA",
 		.dev_name = "1-000a",
 	},
-	{
-		/* sgtl5000 */
-		.supply = "VDDD",
-		.dev_name = "1-000a",
-	},
 };
 
 static struct regulator_consumer_supply vvideo_consumers[] = {
@@ -215,6 +210,7 @@ static struct regulator_init_data vpll_init = {
 		.max_uV = mV_to_uV(1800),
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
 		.boot_on = 1,
+		.always_on = 1,
 	}
 };
 
@@ -225,6 +221,7 @@ static struct regulator_init_data vusb2_init = {
 		.max_uV = mV_to_uV(2775),
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
 		.boot_on = 1,
+		.always_on = 1,
 	}
 };
 
@@ -234,7 +231,6 @@ static struct regulator_init_data vvideo_init = {
 		.min_uV = mV_to_uV(2775),
 		.max_uV = mV_to_uV(2775),
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
-		.always_on = 1,
 		.apply_uV =1,
 	},
 	.num_consumer_supplies = ARRAY_SIZE(vvideo_consumers),
@@ -285,6 +281,7 @@ static struct regulator_init_data vgen2_init = {
 		.min_uV = mV_to_uV(1200),
 		.max_uV = mV_to_uV(3150),
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
+		.always_on = 1,
 	}
 };
 
@@ -294,6 +291,7 @@ static struct regulator_init_data vgen3_init = {
 		.min_uV = mV_to_uV(1800),
 		.max_uV = mV_to_uV(2900),
 		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE,
+		.always_on = 1,
 	}
 };
 
@@ -395,6 +393,8 @@ static int mc13892_regulator_init(struct mc13892 *mc13892)
 	mc13892_register_regulator(mc13892, MC13892_GPO3, &gpo3_init);
 	mc13892_register_regulator(mc13892, MC13892_GPO4, &gpo4_init);
 
+	regulator_has_full_constraints();
+
 	return 0;
 }
 
-- 
1.5.4.4

